### PR TITLE
feat(replay): add viewed_by_ids to replay search fields

### PR DESCRIPTION
**Pull request by aliu39**
Follow up from https://github.com/getsentry/sentry/issues/64924

Before/after:
Before

Screenshot 2024-04-19 at 3 32 34 PM

Followed up with docs PR for searchable properties

[Trigger: Revert]

### CODE REVIEW

File: static/app/utils/fields/index.ts [hunk 1]
```diff
@@ -1362,6 +1364,11 @@ const REPLAY_FIELD_DEFINITIONS: Record<ReplayFieldKey, FieldDefinition> = {
     kind: FieldKind.FIELD,
     valueType: FieldValueType.STRING,
   },
+  [ReplayFieldKey.VIEWED_BY_ID]: {
+    desc: t('Sentry user id(s) who have seen this Replay'),
```
Comment (michellewzhang) [inline static/app/utils/fields/index.ts:1368]: this is a super nit but looks like replay isn't capitalized in the other descriptions 
```suggestion
    desc: t('Sentry user ID(s) who have seen this replay'),
```

**Review by michellewzhang — APPROVED** [review]

File: static/app/utils/fields/index.ts (hunk 1 above)
Comment (aliu3ntry) [reply static/app/utils/fields/index.ts:1368]: oh it is for URLS, but nowhere else

Comment (michellewzhang) [reply static/app/utils/fields/index.ts]: oh i totally missed that! good catch, thanks for updating both

**Comment by aliu3ntry** [comment]
Backend PR for seen_by_id: https://github.com/getsentry/sentry/pull/69370
The two aren't directly dependent -- if we merge this first, seen_by_id will show up in the typeahead but it'll be an invalid search param. Long as they merge close to the same time it should be fine

File: static/app/utils/fields/index.ts [hunk 2]
```diff
@@ -1357,11 +1361,21 @@ const REPLAY_FIELD_DEFINITIONS: Record<ReplayFieldKey, FieldDefinition> = {
     kind: FieldKind.FIELD,
     valueType: FieldValueType.STRING,
   },
+  [ReplayFieldKey.SEEN_BY_ID]: {
+    desc: t('Sentry user id(s) who have seen this Replay'),
```
Comment (michellewzhang) [inline static/app/utils/fields/index.ts:1365]:
```suggestion
    desc: t('Sentry user ID(s) who have seen this Replay'),
```

**Comment by michellewzhang** [comment]
do we need both search keys if they're doing the same thing? should we expose just one in the UI?

**Comment by aliu3ntry** [comment]
> do we need both search keys if they're doing the same thing? should we expose just one in the UI?

I think we can keep both for now and come back later if needed. "viewed" matches the API while "seen" matches the UI. I only documented seen_by, while mentioning the alias: https://github.com/getsentry/sentry-docs/pull/9764

**Comment by aliu3ntry** [comment]
trigger:revert

**Comment by aliu3ntry** [comment]
There's no point exposing this in the search bar since regular users can't lookup user ids. Will replace with a new field after merging backend PR https://github.com/getsentry/sentry/pull/70967

**Pull request merged**